pandas/core/generic.py

```diff
@@ -1084,6 +1084,10 @@ def to_hdf(self, path_or_buf, key, **kwargs):
                        / selecting subsets of the data
         append   : boolean, default False
             For Table formats, append the input data to the existing
+        data_columns : list of columns, or True, default None
+            This will create additional indexed columns for on-disk queries,
+            by default only 'index' and 'columns' are indexed. True will index
+            all columns.
```

Comment:
I think in this case it might not be easy to do this because someone is doing:

```HDFStore.append``, there isn't any object at that point, so you can just give an example e.g. ``Series -> 'index' is defined, DataFrame -> 'index' and 'columns' ,Panel -> 'items', 'major_axis','minor_axis'`` (put them in bullet points).